pkg/neonvm/ipam/metrics.go

```diff
@@ -0,0 +1,79 @@
+package ipam
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/neondatabase/autoscaling/pkg/util"
+)
+
+type IPAMMetrics struct {
+	ongoing  prometheus.GaugeVec
+	duration prometheus.HistogramVec
+}
+
+const (
+	IPAMAcquire = "acquire"
+	IPAMRelease = "release"
+	IPAMCleanup = "cleanup"
+)
+
+const (
+	IPAMSuccess = "success"
+	IPAMFailure = "failure"
+	IPAMUnknown = "unknown"
+)
+
+func NewIPAMMetrics(reg prometheus.Registerer) *IPAMMetrics {
+	// Copied bucket values from controller runtime latency metric. We can
+	// adjust them in the future if needed.
+	buckets := []float64{
+		0.005, 0.01, 0.025, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+		1.25, 1.5, 1.75, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5, 6, 7, 8, 9, 10, 15, 20, 25, 30, 40, 50, 60,
+	}
```

Comment:
suggestion: Let's tune this a little bit? It's 40 buckets × 3 operations × 3 results, and I bet a ton of the values are unneeded. E.g. what if we just do powers of 10 for now, and iterate on the bucket values from tests on staging?

(thinking about this because e.g. VM startup time buckets are wildly off but haven't been fixed)

Not a huge deal, you're welcome to ignore if you like. Just thought I'd mention it

Reply:
Changed to this:
```
buckets := []float64{
		0.01, 0.05, 0.1, 0.25, 0.5, 1, 2, 3, 5, 10, 30,
}
```